Add Highlight feature for the selected video on the 'Individual Courses Page' #252

samad-yar-khan opened the pull request:
**ISSUE 252**

fixes #252 

<!-- Please Mention the issue number as ISSUE #(Issue Number)
Example:
fixes #1
-->

**Describe the changes you've made**

- I have added a 'videoPlaying' property to our state on the the individualCourses page .
- This property will take the ID of the current video selected/played but for now the index of the video from the list is being stored here as it is same as the key for the component as of now .
- Initial value of the 'videoPlaying' property is set to -1 for now , indicating no video selected but later this can be changed to the last video being played by the user.
- Pressing/Clicking on a video from the List on the individualCourses page triggers a playVideo function which takes the key/index of the video which has been selected and resets the state by setting the 'videoPlaying' property equal to the selected videos key/index.
- At the time of rendering of Videos List items in the List , if the video list item being rendered is the one which is being played , then a separate 'playVideo' class and style is added to it using ternary operator and conditional rendering .

<!-- A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know! -->

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

 Unusual Behaviour : `NA`

<!-- A clear and concise description of it. -->

**Additional context (OPTIONAL)**

Page after adding the highlight feature .
Here the first video is selected .
![252](https://user-images.githubusercontent.com/70485812/115483010-8f0c1200-a26d-11eb-8cee-6d89690693d9.PNG)

<!-- Add any other context or screenshots about the feature request here. -->

**Checklist**

<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**
Preview: http://localhost:3000/mycourses/:coursename


<!-- Please add deployed link to here text >

samad-yar-khan commented:
@Abhishek-kumar09 Sorry for the delay . Can you please review my pull request ?

samad-yar-khan commented:
@Abhishek-kumar09 Sir , Can you please review my pull request  ?

Abhishek-kumar09 commented on src/pages/CoursePageViewWithVideo/IndividualCourse.js:
```diff
@@ -98,6 +103,10 @@ export default function CoursePageViewWithVideo(props) {
   };
 
+  const playVideo = (index) => {
+    setState({ ...state, vidPlaying: index });
```
you are using `vidPlaying` here, but `videoPlaying` in the state on line 92

samad-yar-khan commented:
@Abhishek-kumar09 My bad , I changed the property name in the last but didn't make changes in the places where I used it. Thankyou for pointing this out .  Changed all occurrences of vidPlaying property to videoPlaying in the latest commit . Seems to be working fine . Can you please review once again ?

Abhishek-kumar09 commented on src/pages/CoursePageViewWithVideo/IndividualCourse.js:
```diff
@@ -172,7 +181,11 @@ export default function CoursePageViewWithVideo(props) {
             {[1, 2, 3, 4, 5].map((item, index) => {
               return (
-                <ListItem key={index}>
+                <ListItem
```
Since it is clickable, add `cursor: pointer` to the list items.

samad-yar-khan commented:
@Abhishek-kumar09 I have changed the cursor to pointer for the List Items .

Abhishek-kumar09 commented on src/pages/CoursePageViewWithVideo/IndividualCourse.js:
```diff
@@ -85,6 +89,7 @@ export default function CoursePageViewWithVideo(props) {
     open: false,
     vertical: "top",
     horizontal: "center",
+    videoPlaying: -1,
```
change it to `playingIndex`

samad-yar-khan commented:
@Abhishek-kumar09  I have made the changes . Have a look and let me know if somethings missing/needs to be changes .

Abhishek-kumar09 commented:
Thanks @samad-yar-khan

samad-yar-khan commented:
@Abhishek-kumar09  Thanks a lot for reviewing my code multiple times and suggesting all the changes  :)
